lab_1/main.py

```diff
@@ -87,7 +135,29 @@ def get_adjacent_words(tokens: list, word: str, left_n: int, right_n: int) -> li
     right_n = 3
     --> [['man', 'is'], ['dog, 'cat']]
     """
-    pass
+    if not isinstance(tokens, list) or not isinstance(word, str):
+        return []
+    if (not isinstance(left_n, int) and not isinstance(right_n, int)) or \
+            (left_n < 1 and right_n < 1):
+        return []
+    word_pairs = []
+    if not isinstance(left_n, int) or left_n < 1:
+        concordances = get_concordance(tokens, word, left_n, right_n)
```

Comment:
it is not good to call `get_concordance` several times in function. Just call it once and then work with contexts